osu.Game/Graphics/DrawableDate.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) 2007-2018 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using System;
+using Humanizer;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Cursor;
+using osu.Framework.Threading;
+using osu.Game.Graphics.Sprites;
+
+namespace osu.Game.Graphics
+{
+    public class DrawableDate : OsuSpriteText, IHasTooltip
+    {
+        private readonly DateTimeOffset date;
+        private ScheduledDelegate updateTask;
+
+        public DrawableDate(DateTimeOffset date)
+        {
+            AutoSizeAxes = Axes.Both;
+            Font = "Exo2.0-RegularItalic";
+
+            this.date = date.ToLocalTime();
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            updateTime();
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
+            Scheduler.Add(updateTimeWithReschedule);
+        }
+
+        private void updateTimeWithReschedule()
+        {
+            updateTime();
+
+            var diffToNow = DateTimeOffset.Now.Subtract(date);
+
+            double timeUntilNextUpdate = 1000;
+            if (diffToNow.TotalSeconds > 60)
+            {
+                timeUntilNextUpdate *= 60;
+                if (diffToNow.TotalMinutes > 60)
+                {
+                    timeUntilNextUpdate *= 60;
+
+                    if (diffToNow.TotalHours > 24)
+                        timeUntilNextUpdate *= 24;
+                }

```

Comment:
What? Do we really need this sort of optimisation?

Reply:
any reason not to? if there are 100+ rows displayed on a profile this can save a lot of schedules and string construction/comparisons per second.